google/cloud/storage/client.cc

```diff
@@ -268,6 +268,39 @@ credentials, as described in:
   return std::move(os).str();
 }
 
+StatusOr<std::string> Client::SignPolicyDocument(
+    internal::PolicyDocumentRequest const& request) {
+  auto base_credentials = raw_client()->client_options().credentials();
+  auto credentials = dynamic_cast<oauth2::ServiceAccountCredentials<>*>(
+      base_credentials.get());
+
+  if (credentials == nullptr) {
+    return Status(StatusCode::kInvalidArgument,
+                  R"""(The current credentials cannot be used to sign URLs.
+Please configure your google::cloud::storage::Client to use service account
+credentials, as described in:
+https://cloud.google.com/storage/docs/authentication
+)""");
+  }
+
+  auto result = credentials->SignString(request.StringToSign());
+  if (!result.first.ok()) {
+    return result.first;
+  }
+
+  internal::CurlHandle curl;
+  std::string signature = result.second;
+
+  std::ostringstream os;
+  os << "https://storage.googleapis.com/" << request.bucket_name();
+  os << "?GoogleAccessId=" << credentials->client_id()
+     << "&Expires=" << request.expiration_time_as_seconds().count()
```

Comment:
I use `expiration_time_as_seconds()` like V2SignURL but maybe this needs to be a timestamp instead.